File: backend/adapters/src/main/java/br/com/unifor/adapters/dto/ProfessorDTO.java
package br.com.unifor.adapters.dto;

import lombok.*;
import lombok.experimental.SuperBuilder;

@Getter
@Setter
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
public class ProfessorDTO extends AuditableDTO {
    private String nome;
    private String email;
    private String area;
    private Boolean coordenador;
    private UsuarioDTO usuario;
}

File: frontend/unifor-frontend/src/app/features/admin/pages/professores/professor.dto.ts
// admin/pages/professores/professor.dto.ts
import { UsuarioDTO } from '../usuario/usuario.dto';

export interface ProfessorDTO {
  identifier?: number;
  nome: string;
  email: string;
  area: string;
  coordenador?: boolean;
  usuario: UsuarioDTO;
  createdOn?: string;
  updatedOn?: string;
}
